smac/intensification/abstract_racer.py

```diff
@@ -95,14 +101,74 @@ def __init__(self, tae_runner: ExecuteTARun,
         else:
             self.instance_specifics = instance_specifics
 
-    def intensify(self, challengers: typing.List[Configuration],
+        # general attributes
+        self._min_time = 10 ** -5
+        self._num_run = 0
+        self._chall_indx = 0
+        self._ta_time = 0
+
+    def intensify(self, challengers: typing.Optional[typing.List[Configuration]],
+                  optimizer: typing.Optional['SMBO'],
                   incumbent: Configuration,
                   run_history: RunHistory,
                   aggregate_func: typing.Callable,
                   time_bound: float = float(MAXINT),
                   log_traj: bool = True) -> typing.Tuple[Configuration, float]:
+
         raise NotImplementedError()
 
+    def next_challenger(self, challengers: typing.Optional[typing.List[Configuration]],
+                        optimizer: typing.Optional['SMBO'],
+                        n_chall: int,
+                        run_history: RunHistory,
+                        repeat_configs: bool = True) -> Configuration:
+        """ Retuns the next challenger to use in intensification
+        If challenger is None, then optimizer will be used to generate the next challenger
+
+        Parameters
+        ----------
+        challengers : typing.List[Configuration]
+            promising configurations
+        optimizer : SMBO
```

Comment:
Could you please use the full classpath for objects? Then, they'll be rendered in the documentation.